src/main/java/dev/openfeature/sdk/EventHandling.java

```diff
@@ -0,0 +1,17 @@
+package dev.openfeature.sdk;
+
+import java.util.function.Consumer;
+
+/**
+ * Interface for attaching event handlers.
+ */
+public interface EventHandling<T> {
+    
+    T onProviderReady(Consumer<EventDetails> handler);
+    
+    T onProviderConfigurationChanged(Consumer<EventDetails> handler);
+    
+    T onProviderError(Consumer<EventDetails> handler);
+    
+    T onProviderStale(Consumer<EventDetails> handler);
```

Comment:
@lopitz you suggested these kinds of names... except I think you suggested "doOn..." not "on...". Do you have a preference? "doOn..." struck me as a bit redundant, but I don't feel too strongly about it.

Reply:
`doOn...` was just an example. i'm totally fine with these names.